src/internal/SvelteComponent.js

```diff
@@ -0,0 +1,59 @@
+import { schedule_update, flush } from './scheduler';
+
+export class SvelteComponent {
+	constructor(options) {
+		this.$$get_state = this.$$init(
+			fn => this.$$inject_props = fn,
+			fn => this.$$inject_refs = fn,
+			key => this.$$make_dirty(key)
+		);
+
+		this.$$dirty = null;
+
+		if (options.props) {
+			this.$$inject_props(options.props);
+		}
+
+		this.$$fragment = this.$$create_fragment(this, this.$$get_state());
+
+		if (options.target) {
+			this.$$mount(options.target);
+			flush();
+		}
+	}
+
+	$on(eventName, callback) {
+
+	}
+
+	$destroy() {
+		this.$$destroy(true);
+	}
+
+	$$make_dirty(key) {
+		if (!this.$$dirty) {
+			schedule_update(this);
+			this.$$dirty = {};
+		}
+		this.$$dirty[key] = true;
+	}
+
+	$$mount(target, anchor) {
+		this.$$fragment.c();
+		this.$$fragment.m(target, anchor);
+	}
+
+	$$set(key, value) {
```

Comment:
So this doesn't get lost: I had suggested in chat renaming this to `$set` and making it part of the component's public API.